users/models.py

```diff
@@ -1,3 +1,385 @@
+
+    @classmethod
+    def get_by_id(cls, user_id):
+        """
+        Retrieve a user by their ID.
+
+        Args:
+            user_id (int): The ID of the user.
+
+        Returns:
+            User or None: The user instance if found, else None.
+        """
+        return cls.objects.filter(user_id=user_id).first()
+
+    @classmethod
+    def get_by_email(cls, email):
+        """
+        Retrieve a user by their email address.
+
+        Args:
+            email (str): The user's email.
+
+        Returns:
+            User or None: The user instance if found, else None.
+        """
+        return cls.objects.filter(email=email).first()
+
+    @classmethod
+    def create(cls, email, password, first_name, last_name,
+               user_phone=None, title=None, role=None, **other_fields):
+        """
+        Create a new user with field validation.
+
+        Args:
+            email (str): User email.
+            password (str): User password.
+            first_name (str): First name.
+            last_name (str): Last name.
+            user_phone (str, optional): Phone number.
+            title (str, optional): Job title.
+            role (str, optional): User role.
+            **other_fields: Additional fields.
+
+        Raises:
+            ValueError: If validation fails or email already exists.
+
+        Returns:
+            User: The created user instance.
+        """
+        try:
+            validate_email(email)
+        except ValidationError:
+            raise ValueError("Invalid email format")
+
+        if cls.all_objects.filter(email=email).exists():
+            raise ValueError("Email already exists")
+
+        if len(email) > 50:
+            raise ValueError("Email must be 50 characters or fewer")
+        if len(first_name) > 50:
+            raise ValueError("First name must be 50 characters or fewer")
+        if len(last_name) > 50:
+            raise ValueError("Last name must be 50 characters or fewer")
+        if user_phone is not None and len(user_phone) > 20:
+            raise ValueError("User phone must be 20 characters or fewer")
+        if title is not None and len(title) > 100:
+            raise ValueError("Title must be 100 characters or fewer")
+        if role is not None and role not in dict(cls.Role.choices):
+            raise ValueError("Invalid role value")
+        if not password or not isinstance(password, str) or len(password) < 8:
+            raise ValueError("Password must be a string at least 8 characters long")
+
+        user = cls(
+            email=email,
+            first_name=first_name,
+            last_name=last_name,
+            user_phone=user_phone,
+            title=title,
+            role=role or cls.Role.USER,
+            **other_fields
+        )
+        user.set_password(password)
+        try:
+            user.save()
+        except (IntegrityError, DataError) as e:
+            raise ValueError(f"Database error: {e}")
+        return user
```

Comment:
The `create` classmethod is very robust, with thorough validation and clear error messages. This should help prevent bad data from entering the database.